javase/src/main/java/com/google/zxing/client/j2se/DecodeWorker.java

```diff
@@ -149,9 +149,40 @@ private static void dumpResult(URI input, Result... results) throws IOException
 
     if (config.brief) {
       System.out.println(uri + ": Success");
+    } else if (config.json) {
+      StringWriter output = new StringWriter();
+      output.write("[");
+      for (int resultIndex = 0; resultIndex < results.length; resultIndex++) {
+        Result result = results[resultIndex];
+        ParsedResult parsedResult = ResultParser.parseResult(result);
+        output.write("{");
+          output.write("\"uri\":\"" + uri + "\"");
+          output.write(",");
```

Comment:
I can understand breaking this up a bit but the commas can join the preceding lines